internal/common/util/job.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/G-Research/armada/pkg/api"
+)
+
+func PodSpecFromJob(job *api.Job) *v1.PodSpec {
```

Comment:
I'm sure I've seen this exact function somewhere else in our codebase!

Reply:
Indeed, I took it from here: https://github.com/G-Research/armada/blob/aa7cad0c3ec6acb485161277ff464f43a42732e0/internal/scheduler/legacyscheduler.go#L789

Just moved it to util package, as it could be useful elsewhere